associacao-brasileira-de-normas-tecnicas-ufrgs-initials.csl

```diff
@@ -43,196 +50,291 @@
-      <substitute>
-        <names variable="editor"/>
-        <names variable="translator"/>
-        <choose>
-          <if type="book">
-            <text variable="title" form="short"/>
-          </if>
-          <else-if type="broadcast song motion_picture" match="any">
-            <text variable="title"/>
-          </else-if>
-          <else>
-            <text variable="title" form="short" text-case="uppercase" quotes="false"/>
-          </else>
-        </choose>
-      </substitute>
-    </names>
-  </macro>
   <macro name="access">
-    <choose>
-      <if type="book chapter">
-        <text value="E-book" font-style="italic" suffix="."/>
-      </if>
-    </choose>
+    <!-- Macro utilizada para apresentar a URL do site utilizado e data de acesso -->
     <choose>
       <if variable="DOI" match="any">
-        <text variable="DOI" prefix=" Disponível em: https://doi.org/"/>
-        <group prefix=". Acesso em: " suffix=".">
-          <date variable="accessed">
-            <date-part name="day" suffix=" "/>
-            <date-part name="month" form="short" suffix=". "/>
-            <date-part name="year"/>
-          </date>
+        <!-- Quando houver DOI, utilizá-lo no lugar do link -->
+        <text term="available at" text-case="capitalize-first" prefix=" " suffix=": "/>
+        <text variable="DOI" prefix="https://doi.org/"/>
+        <group>
+          <text term="accessed" text-case="capitalize-first" prefix=". " suffix=" "/>
+          <text term="at" text-case="lowercase" suffix=": "/>
+          <text macro="accessed"/>
         </group>
       </if>
       <else-if variable="URL">
-        <text variable="URL" prefix=" Disponível em: "/>
-        <group prefix=". Acesso em: " suffix=".">
-          <date variable="accessed">
-            <date-part name="day" suffix=" "/>
-            <date-part name="month" form="short" suffix=". "/>
-            <date-part name="year"/>
-          </date>
-        </group>
+        <choose>
+          <if type="book chapter">
+            <!-- Quando for livro, incluir a expressão e-book em itálico -->
+            <text value="E-book" font-style="italic" suffix="."/>
+          </if>
+          <else>
+            <text term="available at" text-case="capitalize-first" prefix=" " suffix=": "/>
+            <text variable="URL"/>
+            <group>
+              <!-- formato de acesso utiliza a macro accessed -->
+              <text term="accessed" text-case="capitalize-first" prefix=". " suffix=" "/>
+              <text term="at" text-case="lowercase" suffix=": "/>
+              <text macro="accessed"/>
+            </group>
+          </else>
+        </choose>
       </else-if>
     </choose>
   </macro>
+  <macro name="accessed">
+    <!-- Macro utilizada para a data de acesso -->
+    <date variable="accessed" suffix=".">
+      <date-part name="day" suffix=" "/>
+      <date-part name="month" form="short" suffix=" "/>
+      <date-part name="year" prefix=" "/>
```

Comment:
don't need the prefix here, either. (The two times you call the macro, there's a space in the preceding suffix, so even without day&month you'd still get a space before year)